coding/python_style_guide.rst

```diff
@@ -12,648 +13,671 @@ To request changes to these standards, please file an :ref:`RFC <decision-making
 
-Rather not:
+However, sentences **should not** be separated by two spaces; a single space is sufficient.
 
-.. code-block:: py
+`This differs from PEP 8 <https://www.python.org/dev/peps/pep-0008/#id29>`__.
 
-   if foo == 'blah': do_blah_thing()
-   for x in lst: total += x
-   while t < 10: t = delay()
+.. _style-guide-py-block-comment-indentation:
 
-Definitely not:
+Block comments SHOULD reference the code following them and SHOULD be indented to the same level
+------------------------------------------------------------------------------------------------
 
-.. code-block:: py
+Block comments generally apply to some (or all) code that follows them, and are indented to the same level as that code.
+Each line of a block comment starts with a ``#`` and a single space (unless it is indented text inside the comment).
 
-   if foo == 'blah': do_blah_thing()
-   else: do_non_blah_thing()
- 
-   try: something()
-   finally: cleanup()
- 
-   do_one(); do_two(); do_three(long, argument,
-                                list, like, this)
-                            
-   if foo == 'blah': one(); two(); three()
+Paragraphs inside a block comment are separated by a line containing a single ``#``.
 
-.. _style-guide-py-string-handling:
+.. _style-guide-py-docstrings:
 
-5. String Handling
-==================
+5. Documentation Strings (docstrings)
+=====================================
 
-.. _style-guide-py-string-5-1:
+Use **Numpydoc** to format the content of all docstrings.
+The page :doc:`../docs/py_docs` authoritatively describes this format.
+It's guide should be treated as an extension of this Python style guide.
 
-String methods SHOULD be used instead of the string module
-----------------------------------------------------------
+See also the :doc:`../docs/rst_styleguide` and the :ref:`rst-formatting-guidelines` section in particular for guidelines on reStructuredText in general.
 
-Use `string methods <https://docs.python.org/2/library/stdtypes.html#string-methods>`_ instead of the :py:mod:`string` module.
-String methods are always much faster and share the same API with unicode strings.
+.. _style-guide-py-docstring-public-api:
 
-.. _style-guide-py-string-5-2:
+Docstrings SHOULD be written for all public modules, functions, classes, and methods
+------------------------------------------------------------------------------------
 
-``.startswith()`` and ``.endswith()`` SHOULD be used to check for prefixes or suffixes
---------------------------------------------------------------------------------------
+Write docstrings for all public modules, functions, classes, and methods.
+See :doc:`../docs/py_docs`.
 
-Use :py:meth:`str.startswith()` and :py:meth:`str.endswith()` instead of string slicing to check for prefixes or suffixes; they are cleaner and less error prone.
+Docstrings are not necessary for non-public methods, but you should have a comment that describes what the method does.
+This comment should appear after the ``def`` line.
```

Comment:
No downside at all—I'm just preserving the original text. I'm in favour of docstrings everywhere, but I've gotten pushback from certain people when advocating it in code reviews...

Flag for another (minor) RFC?